standup/auth0/pipeline.py

```diff
@@ -109,3 +125,52 @@ def login_user(request, user, **kwargs):
     # FIXME(willkg): This is sort of a lie--should we have our own backend?
     user.backend = 'django.contrib.auth.backends.ModelBackend'
     login(request, user)
+
+
+def require_id_token(request, user, user_info, token_info, **kwargs):
+    """Handles domains that require id tokens
+
+    For example, if a user logs in with a mozilla.com email address, they must do so using the LDAP
+    provider. Otherwise we log them out.
+
+    If they logged in using the correct provider, then we capture the ``id_token`` so we can
+    periodically renew it and log them out if it goes bad.
+
+    """
+    # Pull the domain and if it's in the list of domains we need an id_token for, do the id_token
+    # work.
+    domain = user_info['email'].lower().split('@', 1)[1]
+    if domain in settings.AUTH0_ID_TOKEN_DOMAINS:
+        if token_info.get('id_token'):
+            # We have an id_token, so persist it.
+            try:
+                token = IdToken.objects.get(user=user)
+            except IdToken.DoesNotExist:
+                token = IdToken(user=user)
+
+            token.id_token = token_info['id_token']
+            token.expire = datetime.utcnow() + timedelta(seconds=settings.AUTH0_ID_TOKEN_EXPIRY)
+            token.save()
+
+        else:
+            # We don't have an id_token, but should, so return a message and eject.
+            messages.error(
+                request,
+                'You can\'t log in with that email address using the provider you '
+                'used. Please log in with the Oauth2 provider.',
+                fail_silently=True
+            )
+            return HttpResponseRedirect(reverse(settings.AUTH0_SIGNIN_VIEW))
+
+
+def mozilla_auth0_pipeline(**kwargs):
```

Comment:
This makes it much easier for Mozilla sites to have the right pipeline and makes it easier for us to update that pipeline when we do releases.